grafana/resource_datasource_permission.go

```diff
@@ -0,0 +1,173 @@
+package grafana
+
+import (
+	"context"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	gapi "github.com/grafana/grafana-api-golang-client"
+)
+
+func ResourceDatasourcePermission() *schema.Resource {
+	return &schema.Resource{
+
+		Description: `
+* [HTTP API](https://grafana.com/docs/grafana/latest/http_api/datasource_permissions/)
+`,
+
+		CreateContext: UpdateDatasourcePermissions,
+		ReadContext:   ReadDatasourcePermissions,
+		UpdateContext: UpdateDatasourcePermissions,
+		DeleteContext: DeleteDatasourcePermissions,
+
+		Schema: map[string]*schema.Schema{
+			"datasource_id": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				ForceNew:    true,
+				Description: "ID of the datasource to apply permissions to.",
+			},
+			"permissions": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Description: "The permission items to add/update. Items that are omitted from the list will be removed.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"team_id": {
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Default:     0,
+							Description: "ID of the team to manage permissions for.",
+						},
+						"user_id": {
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Default:     0,
+							Description: "ID of the user to manage permissions for.",
+						},
+						"permission": {
+							Type:         schema.TypeString,
+							Optional:     true,
```

Comment:
Even when we have a default value, which in addition is the only possible value, I'd rather have this as a required attribute. Personally I like it explicit better than implicit. WDYT?